@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -577,6 +577,68 @@ export class TextElementsRenderer {
         return memoryUsage;
     }
 
+    /**
+     * Fills the screen with boxes which should block other labels.
+     * @note These boxes have highest priority, so will block all labels.
+     * @param boxes Boxes in screen space which reject other labels.
+     */
+    prepopulateScreen(boxes: Math2D.Box[]) {
+        for (const box of boxes) {
+            this.m_screenCollisions.allocateScreenSpace(box);
+        }
+    }
+
+    /**
+     * Fills the screen with lines which should block other labels.
+     * @note These boxes have highest priority, so will block all labels.
+     * @param lines Lines in screen space which reject other labels.
+     */
+    prepopulateScreenWithLines(lines: LineWithBound[]) {
+        for (const line of lines) {
+            this.m_screenCollisions.allocateIBox(line, true);
+        }
+    }
+
+    /**
+     * Fills the screen with lines projected from world space, see [[Tile.blockingElements]].
+     * @note These boxes have highest priority, so will block all other labels.
+     */
+    prepopulateScreenWithBlockingElements() {
+        const renderList = this.m_mapView.visibleTileSet.dataSourceTileList;
+        renderList.forEach(renderListEntry => {
+            const startLinePointProj = new THREE.Vector3();
+            const endLinePointProj = new THREE.Vector3();
+            for (const tile of renderListEntry.visibleTiles) {
+                for (const pathBlockingElement of tile.blockingElements) {
+                    if (pathBlockingElement.points.length < 2) {
+                        continue;
+                    }
+                    // Project to screen and store it in the ScreenCollisions.
+                    let startLinePoint = pathBlockingElement.points[0];
+                    for (let i = 1; i < pathBlockingElement.points.length; i++) {
```

Comment:
Can we move the allocation of all this objects to the Tile creation and don't do it every frame?
You would only update the screen positions every frame.

Reply:
I would do this in a follow up commit if performance is an issue. I don't want to have this stuck in github too long.

Reply:
If you are ok with that.